lib/validation.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * Performs data validation on an object by verifying that it contains
+ * all required fields specified in a given schema.
+ *
+ * Returns true if the object is valid agianst the schema and false otherwise.
+ */
+exports.validateAgainstSchema = function (obj, schema) {
+  return obj && Object.keys(schema).every(
+    field => !schema[field].required || obj[field]
```

Comment:
I got a problem in assignment 3 where the `admin` field in `UserSchema` is `false`. This evaluates `obj[field]` to `false`, which is a JS falsy value. So, if a schema has a required boolean field and if the provided object has that field being `false`, then `!schema[field].required || obj[field]` will evaluate to `false || false` which will be `false`, which is not correct.

In my assignment 3, I made this change and it worked for boolean field:

    !schema[field].required || obj.hasOwnProperty(field)